internal/allocator.h

```diff
@@ -47,6 +47,13 @@
 #include <malloc.h>
 #define GEMMLOWP_USE_MEMALIGN
 #endif
+// posix_memalign is missing on some 4.1 x86 devices
+#if __ANDROID_API__ == 16
+#if __i386__
```

Comment:
Also see: https://github.com/google/gemmlowp/blob/master/internal/common.h#L58
The "gemmlowp" uniform way to do such a platform check would be
#ifdef GEMMLOWP_X86_32